src/algebra/module_regular.lean

```diff
@@ -132,6 +132,10 @@ not_is_regular_zero_iff.mpr nM
 lemma is_regular_of_smul_eq_one (h : a • s = 1) : is_regular M s :=
 is_regular.of_smul a (by { rw h, exact is_regular_one })
 
+/-- An element of `R` admitting a left inverse is `M`-regular. -/
+lemma is_regular_of_mul_eq_one (h : a * b = 1) : is_regular M b :=
+is_regular.of_smul a (by { rw [smul_eq_mul, h], exact is_regular_one })
```

Comment:
Does this proof work?
```suggestion
is_regular_of_smul_eq_one h
```

Reply:
I do not think that this works...

Reply:
Well in that case, does this work?
```suggestion
is_regular.of_mul a (by { rw [h], exact is_regular_one })
```

Reply:
Not even this...

Reply:
How does it fail?

Reply:
I am not sure, but I think that the lemma that you are referring to is the one below, right?  I think that the proof that you were thinking is the one that is already in the file... maybe!

```lean
/-- An element of `S` admitting a left inverse in `R` is `M`-regular. -/
lemma of_smul_eq_one (h : a • s = 1) : is_smul_regular M s :=
of_smul a (by { rw h, exact one M })
```

Reply:
Or maybe it is this one, that I just golfed:

```
/-- An element of `R` admitting a left inverse is `M`-regular. -/
lemma of_mul_eq_one (h : a * b = 1) : is_smul_regular M b :=
of_mul (by { rw h, exact one M })
```